hs_core/hydroshare/users.py

```diff
@@ -542,6 +545,185 @@ def get_public_groups():
+    """
+    deletes a resource as a favorite resource - not actual deletion of resource (wrapper api for hs_labels app)
+    Args:
+        user: an instance of User object - who wants to delete a resource as a favorite resource
+        resource: an instance of BaseResource which needs to be deleted as a favorite resource
+
+    Returns:
+
+    """
+
+    user.ulabels.unfavorite_resource(resource)
+
+
+def create_resource_folder(user, resource, folder_name):
+    """
+    creates a user specific folder for the specified resource (wrapper api for hs_labels app)
+    Args:
+        user: an instance of User object - who wants to create a folder for the resource
+        resource: an instance of BaseResource for which a folder needs to be created
+        folder_name: name of the folder
+
+    Returns:
+
+    """
+    user.ulabels.file_resource(resource, folder_name)
+
+
+def delete_resource_folder(user, resource):
+    """
+    deletes folder for the specified resource (wrapper api for hs_labels app)
+    Args:
+        user: an instance of User object - who wants to delete folder for the resource
+        resource: an instance of BaseResource for which a folder needs to be deleted
+
+    Returns:
+
+    """
+    user.ulabels.unfile_resource(resource)
+
+
+def add_to_my_resources(user, resource):
+    """
+    marks the specified resource to be included in the user's list of resources (my resources)
+    (wrapper api for hs_labels app)
+    Args:
+        user: an instance of User object - who wants to add the resource to my resources
+        resource: an instance of BaseResource which needs to be added to my resources
+
+    Returns:
+
+    """
+
+    # TODO: Need to find out if we should check if the resource is suitable (meaning the resource is public/discoverable
+    # and user has no explicit access permission) for adding to my resources
+    user.ulabels.claim_resource(resource)
+
+
+def delete_from_my_resources(user, resource):
+    """
+    removes the specified resource from the user's list of resources (my resources) (wrapper api for hs_labels app)
+    Args:
+        user: an instance of User object - who wants to remove the resource from my resources
+        resource: an instance of BaseResource which needs to be removed from my resources
+
+    Returns:
+
+    """
+
+    user.ulabels.unclaim_resource(resource)
+
+
+def get_user_resource_labels(user):
+    """
+    gets a list of user saved labels (strings)
+    Args:
+        user: an instance of User object - whose saved resource labels are needed
+
+    Returns: a list of strings
+
+    """
+
+    return user.ulabels.saved_labels
+
+
+def get_user_resource_used_labels(user):
+    """
+    gets a list of labels (strings) currently used by the user in labeling resources
+    Args:
+        user: an instance of User object - whose saved resource labels are needed
+
+    Returns: a list of strings
+
+    """
+
+    return user.ulabels.user_labels
+
```

Comment:
I agree with this, in the sense that every user has a field user.ulabels. I would rather place any of these utilities into the UserLabels class if they prove useful.